engine/src/main/java/org/pentaho/di/job/entries/copymoveresultfilenames/JobEntryCopyMoveResultFilenames.java

```diff
@@ -0,0 +1,702 @@
+      add_time = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "add_time" ) );
+      SpecifyFormat = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "SpecifyFormat" ) );
+      AddDateBeforeExtension =
+        "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "AddDateBeforeExtension" ) );
+
+      date_time_format = XMLHandler.getTagValue( entrynode, "date_time_format" );
+      action = XMLHandler.getTagValue( entrynode, "action" );
+
+      OverwriteFile = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "OverwriteFile" ) );
+      CreateDestinationFolder =
+        "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "CreateDestinationFolder" ) );
+      RemovedSourceFilename = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "RemovedSourceFilename" ) );
+      AddDestinationFilename =
+        "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "AddDestinationFilename" ) );
+
+    } catch ( KettleXMLException xe ) {
+      throw new KettleXMLException( BaseMessages.getString(
+        PKG, "JobEntryCopyMoveResultFilenames.CanNotLoadFromXML", xe.getMessage() ) );
+    }
+  }
+
+  public void loadRep( Repository rep, IMetaStore metaStore, ObjectId id_jobentry, List<DatabaseMeta> databases,
+    List<SlaveServer> slaveServers ) throws KettleException {
+    try {
+      foldername = rep.getJobEntryAttributeString( id_jobentry, "foldername" );
+      specifywildcard = rep.getJobEntryAttributeBoolean( id_jobentry, "specify_wildcard" );
+      wildcard = rep.getJobEntryAttributeString( id_jobentry, "wildcard" );
+      wildcardexclude = rep.getJobEntryAttributeString( id_jobentry, "wildcardexclude" );
+      destination_folder = rep.getJobEntryAttributeString( id_jobentry, "destination_folder" );
+      nr_errors_less_than = rep.getJobEntryAttributeString( id_jobentry, "nr_errors_less_than" );
+      success_condition = rep.getJobEntryAttributeString( id_jobentry, "success_condition" );
+      add_date = rep.getJobEntryAttributeBoolean( id_jobentry, "add_date" );
+      add_time = rep.getJobEntryAttributeBoolean( id_jobentry, "add_time" );
+      SpecifyFormat = rep.getJobEntryAttributeBoolean( id_jobentry, "SpecifyFormat" );
+      date_time_format = rep.getJobEntryAttributeString( id_jobentry, "date_time_format" );
+      AddDateBeforeExtension = rep.getJobEntryAttributeBoolean( id_jobentry, "AddDateBeforeExtension" );
+      action = rep.getJobEntryAttributeString( id_jobentry, "action" );
+
+      OverwriteFile = rep.getJobEntryAttributeBoolean( id_jobentry, "OverwriteFile" );
+      CreateDestinationFolder = rep.getJobEntryAttributeBoolean( id_jobentry, "CreateDestinationFolder" );
+      RemovedSourceFilename = rep.getJobEntryAttributeBoolean( id_jobentry, "RemovedSourceFilename" );
+      AddDestinationFilename = rep.getJobEntryAttributeBoolean( id_jobentry, "AddDestinationFilename" );
+
+    } catch ( KettleException dbe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)